src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Mac.java

```diff
@@ -279,14 +279,14 @@ protected void engineUpdate(ByteBuffer byteBuffer) {
             if (len <= 0) {
                 return;
             }
-            if (!(byteBuffer instanceof DirectBuffer dByteBuffer)) {
+            if (!(byteBuffer instanceof DirectBuffer)) {
                 super.engineUpdate(byteBuffer);
                 return;
             }
             int ofs = byteBuffer.position();
             NIO_ACCESS.acquireSession(byteBuffer);
             try  {
-                token.p11.C_SignUpdate(session.id(), dByteBuffer.address() + ofs, null, 0, len);
+                token.p11.C_SignUpdate(session.id(), NIO_ACCESS.getBufferAddress(byteBuffer) + ofs, null, 0, len);
```

Comment:
I think I would introduce a long address here too, only to make the line easier to read.